superset-frontend/src/views/CRUD/data/dataset/AddDataset/DatasetPanel/DatasetPanel.tsx

```diff
@@ -94,7 +94,7 @@ const LoaderContainer = styled.div`
   display: flex;
   align-items: center;
   justify-content: center;
-  height: 100%;
+  height: 98%;
```

Comment:
@lyndsiWilliams typically we we see scrolling with something where width / height is set to 100% due to box-sizing logic.  By default CSS does not subtract  padding and border thickness from the value of the parent elements size.  This makes the child set its size to 100% of the parent element plus the child's padding and border dimensions (100% of parent + child's padding and borders making the child larger than the parents visible area).  Getting rid of the scrollbar should be achievable with:
```
height: 100%;
box-sizing: border-box;
```
The CSS `box-sizing` property allows us to include the padding and border in an element's total width and height calculation 

The reason we might want to try this instead of reducing the height to 98% is that 2% of the total height in actual pixels varies on the browser window's height.  The higher the screen resolution and taller the browser window is, the more that 2% will cause an offset of actual vertical center alignment.  For example on a HD screen the offset would be ~ 17px (850px of usable screen after browsers UI. 850 * 0.02 = ~17px).  On a 4k screen the offset could be significantly larger.  In other words as the browser grows in height, the more non-centered aligned the item becomes.  Alternatively, as the screen size reduces it becomes possible that 2% is less than the height - padding+border and scroll bar still ends up appearing.  Using 100% with the box-sizing ensures we remain center aligned and void scroll regardless of browser height and screen resolution.

good explanation with interactive examples of box-sizing here: https://www.w3schools.com/css/css3_box-sizing.asp

Reply:
Oh that makes sense, I didn't know about box-sizing, thank you! But setting height to 100% and box-sizing: border-box did not get rid of the scrollbar on my end. Did it work on your end?

I see the reasoning for wanting to include 100% of the element, but I was thinking it might be acceptable to do 98% height for this since it's just the loading gif that shows in the center, so that 2% will never cut any visual elements off. What do you think?

Reply:
Implemented the border-box fix we discussed on Slack in [`this commit`](https://github.com/apache/superset/pull/22136/commits/11e2409d78b3e9227fa698f942eb999ae1570351).